test/cql-pytest/test_secondary_index.py

```diff
@@ -344,3 +346,104 @@ def test_index_weird_chars_in_col_name(cql, test_keyspace):
             iswordchar = lambda x: str.isalnum(x) or x == '_'
             cleaned_up_column_name = ''.join(filter(iswordchar, magic_path))
             assert index_name == cf + '_' + cleaned_up_column_name + '_idx'
+
+def test_secondary_collection_index(cql, test_keyspace):
```

Comment:
Please add a comment what is being tested in this test (besides the obvious fact that it tests "collection indexing").

If I understand correctly, the idea of this test is to build some indentical random data on two tables tab1 and tab2, and build indexes only on tab1 - and then do identical queries on both of them and expect identical results even though supposedly tab1 uses the index and tab2 doesn't (and also requires "allow filtering" on the query). This should be explained. Moreover, if I understand correctly, instead of just assigning entire collections you deliberately add and remove items from them to see if the MV update code worked as expected (right?).

My "style" is a bit different from this - I would have written many small tests all using the same table with multiple columns (for speed) - designed to exhaustively cover all the various cases supported by the code and not rely on any randomness. Such tests are especially easy to write while you're writing the code - and still have fresh memory of what cases there actually are. But I guess I can accept tests in other styles :-)

Reply:
I commented the test. I prefer this style here, since in my opinion, there are no obvious edge cases here. Randomized tests can help against unpredicted edge cases, which might happen in a code that has relatively high cyclomatic complexity. But if you insist, I might rewrite the test.

Reply:
No, I don't insist. Just please just try to keep this test under one second, or a few seconds if you absolutely must. Try running test/cql-pytest/run for all tests, and see how quickly the current tests take. None of them come even close to 30 seconds. Even 1 second is considered a very slow test ;-)

Also thanks for adding a comment, but I would also mention in the comment the fact that instead of just writing random collections up-front, you deliberately build the collections incrementally with various adds and deletes of individual items, to see that these incremental operations update the materialized view correctly.